### PR TITLE
MGMT-3063 Add Env variable to enable/disable CVO wait 

### PR DESCRIPTION
**Pull request by nmagnezi**
To be followed up with a proper config flag.

### CODE REVIEW

**Comment by nmagnezi** [comment]
/hold
testing

File: src/assisted_installer_controller/assisted_installer_controller_test.go [hunk 1]
```diff
@@ -314,9 +314,10 @@ var _ = Describe("installer HostRoleMaster role", func() {
 
 	Context("validating AddRouterCAToClusterCA", func() {
 		conf := ControllerConfig{
-			ClusterID:        "cluster-id",
-			URL:              "https://assisted-service.com:80",
-			OpenshiftVersion: "4.7",
+			ClusterID:           "cluster-id",
+			URL:                 "https://assisted-service.com:80",
+			OpenshiftVersion:    "4.7",
+			CheckClusterVersion: true,
```
Comment (ronniel1) [inline src/assisted_installer_controller/assisted_installer_controller_test.go:320]: Shouldn't we also check a case where this is false?

**Comment by YuviGold** [comment]
@yevgeny-shnaidman Does it replace the code in test-infra?

**Comment by yevgeny-shnaidman** [comment]
> @yevgeny-shnaidman Does it replace the code in test-infra?

more like changes it: now instead of waiting to use wait-for-cvo flag after installation, we need to pass it as part of the assisted-service deployment

File: src/assisted_installer_controller/assisted_installer_controller_test.go (hunk 1 above)
Comment (nmagnezi) [reply src/assisted_installer_controller/assisted_installer_controller_test.go:320]: Yes, we do.
Done.

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ type ControllerConfig struct {
 	CACertPath           string `envconfig:"CA_CERT_PATH" required:"false" default:""`
 	Namespace            string `envconfig:"NAMESPACE" required:"false" default:"assisted-installer"`
 	OpenshiftVersion     string `envconfig:"OPENSHIFT_VERSION" required:"true"`
+	CheckClusterVersion  bool   `envconfig:"CHECK_CLUSTER_VERSION" required:"false" default:"false"`
```
Comment (gamli75) [inline src/assisted_installer_controller/assisted_installer_controller.go:55]: I think the name is a little bit confusing and not clear - maybe it should be called: WAIT_FOR_CLUSTER_VERSION

**Review by gamli75 — COMMENTED** [review]
Maybe we should add it to openshift template so we will be able to change it without changing the code - @ronniel1 WDYT?

File: src/assisted_installer_controller/assisted_installer_controller.go (hunk 2 above)
Comment (nmagnezi) [reply src/assisted_installer_controller/assisted_installer_controller.go:55]: np problem, I'll change that.

**Comment by nmagnezi** [comment]
> Maybe we should add it to openshift template so we will be able to change it without changing the code - @ronniel1 WDYT?

@gamli75 the plan to have this env variable in assisted-service configMap, forward it to assisted-installer to render the controller ConfigMap with the same value. 
Will happen in follow-up PRs.

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 3]
```diff
@@ -45,13 +45,14 @@ var WaitTimeout = 2 * time.Hour
...
-	Namespace            string `envconfig:"NAMESPACE" required:"false" default:"assisted-installer"`
-	OpenshiftVersion     string `envconfig:"OPENSHIFT_VERSION" required:"true"`
+	ClusterID             string `envconfig:"CLUSTER_ID" required:"true" `
+	URL                   string `envconfig:"INVENTORY_URL" required:"true"`
+	PullSecretToken       string `envconfig:"PULL_SECRET_TOKEN" required:"true"`
+	SkipCertVerification  bool   `envconfig:"SKIP_CERT_VERIFICATION" required:"false" default:"false"`
+	CACertPath            string `envconfig:"CA_CERT_PATH" required:"false" default:""`
+	Namespace             string `envconfig:"NAMESPACE" required:"false" default:"assisted-installer"`
+	OpenshiftVersion      string `envconfig:"OPENSHIFT_VERSION" required:"true"`
+	waitForClusterVersion bool   `envconfig:"WAIT_FOR_CLUSTER_VERSION" required:"false" default:"false"`
```
Comment (tsorya) [inline src/assisted_installer_controller/assisted_installer_controller.go:55]: upper case

Comment (nmagnezi) [reply src/assisted_installer_controller/assisted_installer_controller.go]: yup

**Pull request merged**